core/services/workflows/engine.go

```diff
@@ -807,6 +858,95 @@ func (e *Engine) deregisterTrigger(ctx context.Context, t *triggerCapability, tr
 	return nil
 }
 
+func (e *Engine) isWorkflowFullyProcessed(state store.WorkflowExecution) (bool, string, error) {
+	statuses := map[string]string{}
+	// we need to first propagate the status of the errored status if it exists...
+	err := e.workflow.walkDo(workflows.KeywordTrigger, func(s *step) error {
```

Comment:
do these funcs need to be anonymous? unit testing `isWorkflowFullyProcessed` seems like it could be cleaner if you could call into these

Reply:
they can be named, I see no issues there.